File: app/src/main/java/workshop1024/com/xproject/model/subinfo/SubInfo.java
package workshop1024.com.xproject.model.subinfo;

public class SubInfo {
    //Info的id
    private String mInfoId;

    //Info图标URL
    private String mIconUrl;
    //Info名称
    private String mName;
    //Info未阅读新闻数量
    private String mUnreadCount;

    public SubInfo(String infoId, String iconUrl, String name, String unreadCount) {
        mInfoId = infoId;
        mIconUrl = iconUrl;
        mName = name;
        mUnreadCount = unreadCount;
    }

    public String getInfoId() {
        return mInfoId;
    }

    public void setInfoId(String infoId) {
        mInfoId = infoId;
    }

    public String getIconUrl() {
        return mIconUrl;
    }

    public void setIconUrl(String iconUrl) {
        mIconUrl = iconUrl;
    }

    public String getName() {
        return mName;
    }

    public void setName(String name) {
        mName = name;
    }

    public String getUnreadCount() {
        return mUnreadCount;
    }

    public void setUnreadCount(String unreadCount) {
        mUnreadCount = unreadCount;
    }

    @Override
    public String toString() {
        return "SubInfo{" +
                "mInfoId='" + mInfoId + '\'' +
                ", mIconUrl='" + mIconUrl + '\'' +
                ", mName='" + mName + '\'' +
                ", mUnreadCount='" + mUnreadCount + '\'' +
                '}';
    }
}

File: home/src/main/java/workshop1024/com/xproject/home/model/subinfo/SubInfo.kt
package workshop1024.com.xproject.home.model.subinfo

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "subinfoes")
data class SubInfo(//Info的id var
        @PrimaryKey @ColumnInfo(name = "infoId") var mInfoId: String, //Info图标URL
        @ColumnInfo(name = "infoType") var mInfoType: String,//Info类型
        @ColumnInfo(name = "iconUrl") var mIconUrl: String, //Info名称
        @ColumnInfo(name = "name") var mName: String, //Info未阅读新闻数量
        @ColumnInfo(name = "unreadCount") var mUnreadCount: String) {

    companion object {
        const val SUBSCRIBE_TYPE: String = "1"
        const val TAG_TYPE: String = "2"
        const val FILTER_TYPE: String = "3"
    }
}